kalite/topic_tools/__init__.py

```diff
@@ -120,14 +122,57 @@ def get_leafed_topics(force=False):
         LEAFED_TOPICS = [topic for topic in get_node_cache()["Topic"].values() if [child for child in topic.get("children", []) if child.get("kind") != "Topic"]]
     return LEAFED_TOPICS
 
+def create_thumbnail_url(thumbnail):
+    if is_content_on_disk(thumbnail, "png"):
+        return settings.CONTENT_URL + thumbnail + ".png"
+    elif is_content_on_disk(thumbnail, "jpg"):
+        return settings.CONTENT_URL + thumbnail + ".jpg"
+    return None
+
 CONTENT          = None
 CACHE_VARS.append("CONTENT")
-def get_content_cache(force=False):
+def get_content_cache(force=False, annotate=False):
     global CONTENT, CONTENT_FILEPATH
 
     if CONTENT is None or force:
         CONTENT = softload_json(CONTENT_FILEPATH, logger=logging.debug, raises=False)
- 
+    
+    if annotate:
+        if settings.HARD_CONTENT_CACHE and not force:
+            content = softload_json(CONTENT_FILEPATH + ".cache", logger=logging.debug, raises=False)
+            if content:
+                CONTENT = content
+                return CONTENT
+        for content in CONTENT.values():
+            languages = []
+            default_thumbnail = create_thumbnail_url(content.get("id"))
+            dubmap = i18n.get_id2oklang_map(content.get("id"))
+            for lang, dubbed_id in dubmap.items():
+                # TODO (rtibbles): Explicitly stamp "mp4" on KA videos in contentload
+                format = content.get("format", "mp4")
+                if is_content_on_disk(dubbed_id, format):
+                    languages.append(lang)
+                    thumbnail = create_thumbnail_url(dubbed_id) or default_thumbnail
+                    content["lang_data_" + lang] = {
+                        "stream": settings.CONTENT_URL + dubmap.get(lang) + "." + format,
+                        "stream_type": "{kind}/{format}".format(kind=content.get("kind", "").lower(), format=format),
```

Comment:
Cool hack. Might be good to (at some point) have this be a lookup table so that we're not stuck with mime type naming for `Kind`? Or just go whole-hog and embrace mimetypes for `Kind` naming, I guess.